src/capture/middlewares/header_middleware/sqlite_header.c

```diff
@@ -595,156 +595,65 @@ int save_packet_statement(sqlite3 *db, struct tuple_packet *tp) {
 }
 
 int init_sqlite_header_db(sqlite3 *db) {
-
-  int rc;
-
   if (db == NULL) {
     log_error("db param is NULL");
     return -1;
   }
 
   log_debug("sqlite autocommit mode=%d", sqlite3_get_autocommit(db));
 
-  rc = check_table_exists(db, "eth");
-
-  if (rc == 0) {
-    log_debug("eth table doesn't exist creating...");
-    if (execute_sqlite_query(db, ETH_CREATE_TABLE) < 0) {
-      log_error("execute_sqlite_query fail");
-      return -1;
-    }
-  } else if (rc < 0) {
-    log_error("check_table_exists fail");
+  if (execute_sqlite_query(db, ETH_CREATE_TABLE) < 0) {
+    log_error("execute_sqlite_query fail");
```

Comment:
Looks good!

It might be worth replacing most of this function with:

```c
  const char* create_table_commands[] = {
    ETH_CREATE_TABLE,
    ARP_CREATE_TABLE,
    IP4_CREATE_TABLE,
    IP6_CREATE_TABLE,
    TCP_CREATE_TABLE,
    UDP_CREATE_TABLE,
    ICMP4_CREATE_TABLE,
    ICMP6_CREATE_TABLE,
    DNS_CREATE_TABLE,
    MDNS_CREATE_TABLE,
    DHCP_CREATE_TABLE,
  };

#ifndef ARRAY_SIZE
#define ARRAY_SIZE(X) sizeof((X))/sizeof((X)[0])
#endif

  for (int i = 0; i < ARRAY_SIZE(create_table_commands); i++) {
    if (execute_sqlite_query(db, create_table_commands[i]) < 0) {
      log_error("execute_sqlite_query fail: %s", create_table_commands[i]);
      return -1;
    }
  }
```

It would definitely make it easier to add/remove more tables (but the error message would need to say exactly which table failed)

Reply:
Agree. Less code is better. Smaller chances for something to fail.

Thanks for the suggestion. I will change it.